lib/util.js

```diff
@@ -64,33 +60,33 @@ export function viewport(obj, options) {
  * @api private
  */
 
-export function save(streams) {
+export async function save(streams) {
 	const files = [];
 
-	function end() {
-		return Promise.all(files.map(file => pify(rimraf, Promise)(file)));
+	async function end() {
+		return await* files.map(file => pify(rimraf, Promise)(file));
 	}
 
-	process.on('SIGINT', () => end().then(process.exit));
+	process.on('SIGINT', async () => process.exit(await end()));
 
-	return Promise.all(streams.map(stream => {
-		return pify(mkdirp, Promise)(this.dest()).then(() => {
-			return new Promise((resolve, reject) => {
-				const dest = path.join(this.dest(), stream.filename);
-				const write = fsWriteStreamAtomic(dest);
+	return await* streams.map(async stream => {
+		return new Promise(async (resolve, reject) => {
+			await pify(mkdirp, Promise)(this.dest());
 
-				files.push(write.__atomicTmp);
+			const dest = path.join(this.dest(), stream.filename);
+			const write = fsWriteStreamAtomic(dest);
 
-				stream.on('warn', this.emit.bind(this, 'warn'));
-				stream.on('error', err => end().then(reject(err)));
+			files.push(write.__atomicTmp);
 
-				write.on('finish', resolve);
-				write.on('error', err => end().then(reject(err)));
+			stream.on('warn', this.emit.bind(this, 'warn'));
+			stream.on('error', err => end().then(reject(err)));
 
-				stream.pipe(write);
-			});
+			write.on('finish', resolve);
+			write.on('error', err => end().then(reject(err)));
+
+			stream.pipe(write);
 		});
-	}));
+	});
 }
```

Comment:
Not really glad about the new implementation of this method. Any improvements are more then welcome.